src/mjolnir/hierarchybuilder.cc

```diff
@@ -308,7 +308,8 @@ void FormTilesInNewLevel(GraphReader& reader,
       // Do we need to force adding edgeinfo (opposing edge could have diff names)?
       // If end node is in the same tile and there is no opposing edge with matching
       // edge_info_offset).
-      bool diff_names = directededge->endnode().tileid() == base_edge_id.tileid() &&
+      bool diff_names = directededge->use() != Use::kTransitConnection &&
```

Comment:
The `OpposingEdgeInfoMatches` was checking the `tile` (note, on local hierarchy!) for the `directededge`'s `endnode` which can be a transit node and then it fails..

I'm often getting confused if `tileid()` also comprises the level or not. But well, it doesn't..

Reply:
Any transit connection (currently? might change that?) has only one name for both directions anyways (AFAICT).